src/Rules/Drupal/PluginManager/PluginManagerSetsCacheBackendRule.php

```diff
@@ -59,7 +59,7 @@ public function processNode(Node $node, Scope $scope): array
                     throw new ShouldNotHappenException();
                 }
                 $cacheKey = $setCacheBackendArgs[1]->value;
-                if (!$cacheKey instanceof Node\Scalar\String_) {
+                if (!$cacheKey instanceof Node\Scalar\String_ && !$cacheKey instanceof Node\Scalar\Encapsed) {
                     continue;
                 }
                 $hasCacheBackendSet = true;
```

Comment:
This whole thing is bad :( It's old code. It needs to be updated for PHPStan's type system. Something like


```php
$type = $scope->getType($setCacheBackendArgs[1]->value);
foreach ($type->getConstantStrings() as ...)
```